provider/azure/azure_discover.go

```diff
@@ -98,26 +99,32 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 func fetchAddrsWithTags(tagName string, tagValue string, vmnet network.InterfacesClient, l *log.Logger) ([]string, error) {
 	// Get all network interfaces across resource groups
 	// unless there is a compelling reason to restrict
-	netres, err := vmnet.ListAll()
+
+	ctx := context.Background()
+	netres, err := vmnet.ListAll(ctx)
+
 	if err != nil {
 		return nil, fmt.Errorf("discover-azure: %s", err)
 	}
 
-	if netres.Value == nil {
+	if len(netres.Values()) == 0 {
 		return nil, fmt.Errorf("discover-azure: no interfaces")
 	}
 
 	// Choose any PrivateIPAddress with the matching tag
 	var addrs []string
-	for _, v := range *netres.Value {
-		id := *v.ID
+	for _, v := range netres.Values() {
+		var id string
```

Comment:
You could default to the index so the log output is somewhat meaningful?

```go
for i, v := range netres.Values() {
    id := strconv.Itoa(i)
    ...
```

Reply:
Ah, in the case of the ID being nil? My worry is that would give false information which might be more confusing. I can set a message of "id not found" or something similar though.